apps/api/src/domain/models/vocabulary-list/entities/vocabulary-list.register-filter-property.spec.ts

```diff
@@ -0,0 +1,137 @@
+                    )
+            );
+
+            it(`should return a new reference to a vocabulary list`, () => {
+                expect(vocabularyListUpdateResult).toBeInstanceOf(VocabularyList); // and not an error
+
+                // ensure (at least at the top level) a new object (clone) has been returned
+                expect(vocabularyListUpdateResult).not.toBe(basicVocabularyList);
+            });
+
+            it(`should add a filter property with the given name`, () => {
+                // i.e., expect we get an object and not undefined
+                expect(newFilterPropertySearchResult).toBeTruthy();
+            });
+
+            it(`should apply the correct type to the new filter`, () => {
+                expect(type).toBe(newFilterPropertyType);
+            });
+
+            it(`should set all of the valid values for the new filter property`, () => {
+                expect(missingAllowedValues).toEqual([]);
+            });
+        });
+    });
+
+    describe(`when the input is invalid`, () => {
+        describe(`when there is already a filter property with the given name`, () => {
+            // we are asserting this update method will not error out (have tested it in the happy path)
+            const vocabularyListWithExistingFilterProperty =
+                basicVocabularyList.registerFilterProperty(
+                    filterPropertyName,
+                    newFilterPropertyType,
+                    allowedValuesWithLabels
+                ) as VocabularyList;
+
+            const vocabularyListUpdateResult =
+                vocabularyListWithExistingFilterProperty.registerFilterProperty(
+                    filterPropertyName,
+                    newFilterPropertyType,
+                    allowedValuesWithLabels
+                );
+
+            it(`should return the expected error`, () => {
+                assertErrorAsExpected(
+                    vocabularyListUpdateResult,
+                    new CannotHaveTwoFilterPropertiesWithTheSameNameError(
+                        filterPropertyName,
+                        basicVocabularyList.id
+                    )
+                );
+            });
+        });
+
+        describe(`when there is an invalid vocabulary list variable definition`, () => {
+            const invalidValuesForCheckbox = [
+                {
+                    // can only be boolean: true \ false
+                    value: 'invalid',
+                    display: 'positive (switch for negative)',
+                },
+                {
+                    value: false,
+                    display: 'negative (switch for positive)',
+                },
+            ];
+
+            it(`should flow through the nested errors`, () => {
+                const result = basicVocabularyList.registerFilterProperty(
+                    'aspect',
+                    DropboxOrCheckbox.checkbox,
+                    // @ts-expect-error TODO fix the types here
```

Comment:
Look into this.

Reply:
Done. It was due to the inconsistency between `LabelAndDisplay` vs. `LabelAndValue`.